### PR TITLE
Cancun spec and config for mainnet

### PR DESCRIPTION
**Pull request by smartprogrammer93**
## Changes

- Cancun spec and config for mainnet

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

Cancun spec and config for mainnet added

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg [hunk 1]
```diff
@@ -6,6 +6,10 @@
     "LogFileName": "mainnet.logs.txt",
     "MemoryHint": 2048000000
   },
+  "TxPool": {
+    "Size": 1024,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:10]: Do we need size, why not default?

File: src/Nethermind/Nethermind.Runner/configs/mainnet_archive.cfg [hunk 2]
```diff
@@ -6,6 +6,10 @@
     "LogFileName": "mainnet_archive.logs.txt",
     "MemoryHint": 4096000000
   },
+  "TxPool": {
+    "Size": 1024,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/mainnet_archive.cfg:10]: Do we need size, why not default?

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg [hunk 3]
```diff
@@ -6,6 +6,10 @@
...
+    "BlobsSupport": "StorageWithReorgs"
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:11]: If we are going to mainnet, I think we should make "StorageWithReorgs" default, and actually set it different only in places where we don't need it.

File: src/Nethermind/Nethermind.Runner.Test/ConfigFilesTests.cs [hunk 4]
```diff
@@ -189,14 +189,14 @@ public void Network_diag_tracer_disabled_by_default(string configWildcard)
             Test<INetworkConfig, bool>(configWildcard, c => c.DiagTracerEnabled, false);
         }
 
-        [TestCase("mainnet", 2048)]
+        [TestCase("mainnet", 1024)]
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner.Test/ConfigFilesTests.cs:192]: why decrease for mainnet? I think we don't want it

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]

**Comment by smartprogrammer93** [comment]
@marcindsobczak what do you think about luk's comments?

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg (hunk 1 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:10]: +1, just use default (2048)

**Comment by marcindsobczak** [comment]
Agree with @LukaszRozmej , we have 2048 as default TxPool size for many years and it works well (we tested 1k, 4k and 8k back then and chose 2k), no need to change it now (especially without benchmarking).

About making `StorageWithReorgs` default - agree as well.
If you @smartprogrammer93 want you can change default value and then adjust every pre-canun config to

```
  "TxPool": {
    "BlobsSupport": "Disabled"
  },
```

But if you want, we can merge it as it is (except 1024 for std TxPool) and I will handle it in separate PR

**Comment by smartprogrammer93** [comment]
@marcindsobczak ill address it in this PR. No problem

File: src/Nethermind/Nethermind.Runner/configs/AuraTest.cfg [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
   "Init": {
-    "ChainSpecPath": "chainspec/AuRaTest.json",    
+    "ChainSpecPath": "chainspec/AuRaTest.json",
     "GenesisHash": "0xa13552e2290059c2736d101e945071a5f3768bc3338c73724184b0606eb8df1d",
     "BaseDbPath": "nethermind_db/aura",
     "LogFileName": "AuraTest.logs.txt"
   },
+  "TxPool": {
+    "BlobsSupport": "Disabled"
+  },
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/AuraTest.cfg:10]: We can skip this one

**Review by LukaszRozmej — APPROVED** [review]

**Review by marcindsobczak — APPROVED** [review]

**Comment by rubo** [comment]
Given the `TxPool.Size: 1024` for Chiado, don't we need the same for Gnosis?

**Comment by smartprogrammer93** [comment]
@rubo i think the idea is testnets usually run on lower spec machines. and that is why we have 1024 for testnets and 2048 for "mainnets"

File: src/Nethermind/Nethermind.Runner/configs/op-sepolia_archive.cfg [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
-    }
+  "Init": {
+    "ChainSpecPath": "chainspec/op-sepolia.json",
+    "GenesisHash": "0x102de6ffb001480cc9b8b548fd05c34cd4f46ae4aa91759393db90ea0409887d",
+    "BaseDbPath": "nethermind_db/op-sepolia-archive",
+    "LogFileName": "op-sepolia-archive.logs.txt",
+    "DisableGcOnNewPayload": false
+  },
+  "TxPool": {
+    "BlobsSupport": "Disabled"
```
Comment (kamilchodola) [inline src/Nethermind/Nethermind.Runner/configs/op-sepolia_archive.cfg:10]: Is it actually valid that OP stack will have blobs disabled? Will there be some separate release for OP chains to support blobs?

**Review by kamilchodola — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]
LGTM

**Pull request merged**